.scripts/ls.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env -S uv run --script --no-project
+
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Output the packages in the repository as a json list.
+
+See the the commandline help for more options.
+"""
+
+from __future__ import annotations
+
+import argparse
+import contextlib
+import dataclasses
+import io
+import json
+import logging
+import pathlib
+import subprocess
+import tarfile
+import tempfile
+
+_REPO_ROOT = pathlib.Path(__file__).parent.parent
+_INTERFACES = _REPO_ROOT / 'interfaces'
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(str(pathlib.Path(__file__).relative_to(_REPO_ROOT)))
+
+
+@dataclasses.dataclass
```

Comment:
Not overly important since it's internal, but I think this probably should be `kw_only`.

Reply:
Good call since we're 3.10+ now.
```suggestion
@dataclasses.dataclass(kw_only=True)
```

Reply:
But as discussed elsewhere, I reckon we'll probably drop this altogether.